KAFKA-19941: Improved property spacing/readability of kafka-features.sh output

rionmonster opened the pull request:
### Description

Closes: [KAFKA-19941](https://issues.apache.org/jira/browse/KAFKA-19941)

The goal of this minor pull request was to improve the existing
readability for the `kafka-features.sh` output which could be difficult
to decipher due to the inadequate spacing between properties. This
change improves that spacing by replacing the previous tab-based spacing
with a fixed-width approach to standardize property alignment.

In addition to these changes all of the affected tests within the
`FeatureCommandTest` suite were updated to ensure all of the existing
tests were passing after the changes were made.

### Discussion and Rationale

These changes were discussed briefly within [this mailing list
thread](https://lists.apache.org/thread/jwcn88o4f4c19x1onymxmkddk3dcsckf),
however the conversation was primarily about if adjusting output would
equate to changing a public interface -- thus requiring a KIP.

While there were several possible options to improve the readability or
to support multiple formats potentially via an output parameter (e.g.
`--output=table` to table-formatted output, similarly with JSON, etc.),
these would be much larger bodies of work. The simplest approach in this
case was to just improve the spacing and allow the end-user to adopt
their preferred transformation from the existing format.

### Example Output

<img width="1381" height="163" alt="kafka-features-alignment-output"
src="https://github.com/user-attachments/assets/0266bbd8-ea91-498f-a0b7-d3f573fdf442"
/>

### Mentions

Tagging @chia7712 for review (as the initial creator of the JIRA)

Reviewers: Chia-Ping Tsai <chia7712@gmail.com>

rionmonster committed:
KAFKA-19941: Improved property spacing/readability of kafka-features.sh output

chia7712 commented on tools/src/main/java/org/apache/kafka/tools/FeatureCommand.java:
```diff
@@ -228,20 +223,12 @@ static String levelToString(String feature, short level) {
             short finalizedLevel = (featureMetadata.finalizedFeatures().get(feature) == null) ? 0 : featureMetadata.finalizedFeatures().get(feature).maxVersionLevel();
             SupportedVersionRange range = featureMetadata.supportedFeatures().get(feature);
-            System.out.printf("Feature: %s\tSupportedMinVersion: %s\tSupportedMaxVersion: %s\tFinalizedVersionLevel: %s\tEpoch: %s%n",
+            System.out.printf("Feature: %s\t\tSupportedMinVersion: %s\t\tSupportedMaxVersion: %s\t\tFinalizedVersionLevel: %s\t\tEpoch: %s%n",
```
Is there a way to align these console outputs based on tab?

rionmonster replied:
Totally! We could introduce fixed width fields to handle the formatting (instead of tabs) which would make it more consistent as seen below:

<img width="1381" height="163" alt="image" src="https://github.com/user-attachments/assets/f86a9586-7511-4cdd-ba25-369899ee44f3" />

chia7712 reviewed (commented):
@rionmonster thanks for this patch. This patch requires a little bit of artistry  😄

rionmonster committed:
KAFKA-19941: Standardized property alignment for kafka-features.sh output

KAFKA-19941: Standardized property alignment for kafka-features.sh output

rionmonster commented:
@chia7712 

I've gone ahead and updated the PR to include the width-based alignment approach and adjusted the associated tests. Let me know what you think and if we need to make any other changes on this front.

chia7712 reviewed (approved):
LGTM

chia7712 commented:
@rionmonster could you rebase code and then fix the new test `testDescribeWithUnstableFeatureVersions`?

rionmonster committed:
KAFKA-19941: Updated property spacing/readability for unstable feature tests in kafka-features.sh

rionmonster commented:
@chia7712 

I've gone ahead and updated the PR to include handling the two newly introduced unstable-related test cases. Let me know if you notice anything else and I'll be happy to address it. 

Thanks again for the feedback!
